rclcpp_lifecycle/src/lifecycle_node_interface_impl.hpp

```diff
@@ -273,63 +277,61 @@ class LifecycleNode::LifecycleNodeInterfaceImpl
     return transitions;
   }
 
-  bool
-  change_state(std::uint8_t lifecycle_transition)
+  rcl_ret_t
+  change_state(std::uint8_t lifecycle_transition, rcl_lifecycle_ret_t & error)
   {
     if (rcl_lifecycle_state_machine_is_initialized(&state_machine_) != RCL_RET_OK) {
-      fprintf(stderr, "%s:%d, Unable to change state for state machine for %s: %s \n",
-        __FILE__, __LINE__, node_base_interface_->get_name(), rcl_get_error_string_safe());
-      return false;
+      RCUTILS_LOG_ERROR("Unable to change state for state machine for %s: %s",
+        node_base_interface_->get_name(), rcl_get_error_string_safe())
+      return RCL_RET_ERROR;
     }
 
     // keep the initial state to pass to a transition callback
     State initial_state(state_machine_.current_state);
 
     uint8_t transition_id = lifecycle_transition;
     if (rcl_lifecycle_trigger_transition(&state_machine_, transition_id, true) != RCL_RET_OK) {
-      fprintf(stderr, "%s:%d, Unable to start transition %u from current state %s: %s\n",
-        __FILE__, __LINE__, transition_id,
-        state_machine_.current_state->label, rcl_get_error_string_safe());
-      return false;
+      RCUTILS_LOG_ERROR("Unable to start transition %u from current state %s: %s",
+        transition_id, state_machine_.current_state->label, rcl_get_error_string_safe())
+      return RCL_RET_ERROR;
     }
 
     rcl_lifecycle_ret_t cb_success = execute_callback(
       state_machine_.current_state->id, initial_state);
+    error = cb_success;
 
     if (rcl_lifecycle_trigger_transition(
         &state_machine_, cb_success, true) != RCL_RET_OK)
     {
-      fprintf(stderr, "Failed to finish transition %u. Current state is now: %s\n",
-        transition_id, state_machine_.current_state->label);
-      return false;
+      RCUTILS_LOG_ERROR("Failed to finish transition %u. Current state is now: %s",
+        transition_id, state_machine_.current_state->label)
+      return RCL_RET_ERROR;
     }
 
     // error handling ?!
     // TODO(karsten1987): iterate over possible ret value
     if (cb_success == RCL_LIFECYCLE_RET_ERROR) {
+      RCUTILS_LOG_WARN("Error occured. Executing error handling.")
```

Comment:
nitpick, I don't think this should be two sentences, instead maybe "Error occurred while doing error handling". (also occurred is misspelled)